### PR TITLE
[#153119879] Upgrade Terraform to latest stable

### PR DESCRIPTION
**Pull request by dcarley**
## What

Upgrade Terraform to latest stable. See individual commits for more information.

## How to review

1. Review the commits individually.
1. Test with other PRs on paas-cf, paas-bootstrap, and paas-release-ci.

## Who can review

Not @dcarley or @46bit

### CODE REVIEW

**Comment by 46bit** [comment]
👍 I'd like to pin the versions. What do you think, @dcarley?

**Comment by dcarley** [comment]
Yep 👍  Are you OK to make the change, @46bit?

**Comment by 46bit** [comment]
@samcrang The container built with the versions 👍

File: terraform/plugin_cache.tf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+provider "aws" {
+  version = "~> 1.6"
```
Comment (samcrang) [inline terraform/plugin_cache.tf]: does it make sense to just use [exact version equality](https://www.terraform.io/docs/configuration/terraform.html#specifying-a-required-terraform-version)?

Comment (46bit) [reply terraform/plugin_cache.tf]: Sure. I've pinned this to the versions that the DockerHub build chose https://hub.docker.com/r/governmentpaas/terraform/builds/bxwtfu2lbnwwqdjt8ktmynw/ :-)

**Pull request merged**